test/module/irohad/multi_sig_transactions/gossip_propagation_strategy_test.cpp

```diff
@@ -143,3 +145,42 @@ TEST(GossipPropagationStrategyTest, ErrorEmitting) {
   auto emitted = subscribe_and_emit(nonstd::nullopt, 1ms, 1, 13);
   ASSERT_EQ(emitted.size(), 0);
 }
+
+/**
+ * @given list of peers and
+ *        strategy that emits two peers
+ * @when strategy emits more than peers available
+ * @then ensure that there's been emitted peers
+ */
+TEST(GossipPropagationStrategyTest, MultipleSubsEmission) {
+  auto peers_size = 10, amount = 2, take = 10;
+  constexpr auto threads = 10;
+  std::vector<std::string> peersId;
+  PropagationData peers = generate(peersId, peers_size);
+
+  std::thread ths[threads];
+  PropagationData result[threads];
+  auto range = boost::irange(0, threads);
+
+  auto query = std::make_shared<MockPeerQuery>();
+  EXPECT_CALL(*query, getLedgerPeers()).WillRepeatedly(testing::Return(peers));
+  GossipPropagationStrategy strategy(query, 1ms, amount);
+
+  std::transform(range.begin(), range.end(), std::begin(ths), [&](auto i) {
+    return std::thread([ =, &res = result[i], &strategy ] {
```

Comment:
`[ =, &res = result[i], &strategy ]` this code looks not so obvious. Is it true that all context captured by value except the strategy?

Reply:
except the `strategy` and `result[i]`, actually

Reply:
In this case, need to write about reasons for this behavior.